### PR TITLE
[MAISTRA-624] Use basic HTTP auth for internal communication

### PR DESCRIPTION
**Pull request by jwendell**
Between kiali, grafana, jaeger and prometheus.
    
This way we can get rid of cluster role bindings.
    
We create an htpasswd secret unconditionally on the beginning of the control
plane installation, configure all oauth proxies to use that htpasswd
file as another form of authentication and finally configure Kiali
CR to use HTTP basic auth to communicate with those services.

### CODE REVIEW

**Comment by jwendell** [comment]
Jaeger is still missing in this PR

**Comment by jpkrohling** [comment]
Jaeger Operator PR: https://github.com/jaegertracing/jaeger-operator/pull/573

**Comment by objectiser** [comment]
@jwendell FYI - the new Jaeger images are listed [here](https://issues.jboss.org/browse/TRACING-633?focusedCommentId=13765460&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13765460), and should hopefully be approved by our QE tomorrow.

Note that the image names have changed.

File: helm/istio/charts/kiali/templates/kiali-cr.yaml [hunk 1]
```diff
@@ -36,22 +36,28 @@ spec:
     grafana:
       auth:
         ca_file: "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
-        type: "bearer"
-        use_kiali_token: true
+        type: "basic"
+        use_kiali_token: false
+        username: "internal"
+        password: ""
```
Comment (rcernich) [inline helm/istio/charts/kiali/templates/kiali-cr.yaml:42]: There's no way to specify a secret for this?

Comment (jwendell) [reply helm/istio/charts/kiali/templates/kiali-cr.yaml:42]: As discussed on IRC: No, not at the moment.

**Review by rcernich — APPROVED** [review]

**Review by dmitri-d — APPROVED** [review]
lgtm

**Comment by jwendell** [comment]
@objectiser @jpkrohling Are you going to release a new community image with those changes in?

Currently we are installing jaeger operator using the [v1.13.1 tag](https://github.com/Maistra/istio-operator/#installing-the-jaeger-operator), which points to the image jaegertracing/jaeger-operator:1.13.1.

On my tests, using an image generated by @jpkrohling , I was only able to deploy the jaeger operator if using latest CR's from master (instead of using the v1.13.1 GitHub tag). So, I believe that just pushing a new image to docker hub under the same [floating] 1.13.1 tag is not enough.

**Comment by jpkrohling** [comment]
We have a `master` tag that represents the current state of master. Our plan is to release a new operator version as soon as a new Jaeger version is released, as we usually do. If there's a need in the community for an operator release before 1.14 is released, we can certainly do so.

**Comment by jwendell** [comment]
Manually merged because mergify is dead

**Pull request merged**